result/ospfv2lsa.yaml

```diff
@@ -0,0 +1,349 @@
+          type: integer
+          format: uint32
+          x-field-uid: 3
+        age:
+          description: >-
+            The time since the LSA's generation in seconds.
+          type: integer
+          format: uint32
+          x-field-uid: 4
+        option_bits:
+          description: >-
+            The optional bits.
+          type: integer
+          format: uint32
+          x-field-uid: 5
+    Ospfv2.Link:
+      description: >-
+        Generic attributes used to identify links within OSPFv2.
+      type: object
+      properties:
+        type:
+          description: >-
+            The data associated with the link type. The value is
+            dependent upon the subtype of the LSA.
+            - point_to_point: The LSA represents a point-to-point connection to another router.
+            - transit: The LSA represents a connection to a transit network.
+            - stub: The LSA represents a connection to a stub network.
+            - virtual: The LSA represents a virtual link connection.
+          type: string
+          x-enum:
+            point_to_point:
+              x-field-uid: 1
+            transit:
+              x-field-uid: 2
+            stub:
+              x-field-uid: 3
+            virtual:
+              x-field-uid: 4
+          x-field-uid: 1
+        id:
+          description: |-
+           The identifier for the link specified. The value of the link
+           identifier is dependent upon the type of the LSA.
+          type: string
+          format: ipv4
+          x-field-uid: 2
+        data:
+          description: |-
+            The data associated with the link type. The value is
+            dependent upon the subtype of the LSA. When the connection is
+            to a stub network it represents the mask; for p2p connections
+            that are unnumbered it represents the ifIndex value of the
+            router's interface; for all other connections it represents
+            the local system's IP address.
+          type: string
+          format: ipv4
+          x-field-uid: 3
+        metric:
+          description: |-
+            The data associated with the link type. The value is
+            dependent upon the subtype of the LSA. When the connection is
+            to a stub network it represents the mask; for p2p connections
+            that are unnumbered it represents the ifIndex value of the
+            router's interface; for all other connections it represents
+            the local system's IP address.
+          type: integer
+          format: uint32
+          x-field-uid: 4
+    Ospfv2.LsaNeighbor:
+      description: >-
+        Generic attributes used to identify OSPFv2 Network LSA neighbor(s).
+      type: object
+      properties:
+        neighbor_router_id:
+          description: >-
+            Neighbor's Router ID for the link.
+          type: string
+          format: ipv4
+          x-field-uid: 1
+        network_mask:
```

Comment:
I think the network_mask should be common and it neighbor_router_id can simply be a ipv4 array in the network_lsa.
Refer to https://www.rfc-editor.org/rfc/rfc2328.html#appendix-A.4.3
or the example at 
https://www.rfc-editor.org/rfc/rfc2328.html#page-135

Reply:
Done.